pkg/ride/converters.go

```diff
@@ -1355,13 +1355,20 @@ func signatureToProofs(sig *crypto.Signature) rideList {
 
 func proofs(proofs *proto.ProofsV1) rideList {
 	r := make(rideList, 8)
-	pl := len(proofs.Proofs)
-	for i := 0; i < 8; i++ {
-		if i < pl {
-			r[i] = rideBytes(common.Dup(proofs.Proofs[i].Bytes()))
-			continue
+	if proofs != nil {
+		proofsLen := len(proofs.Proofs)
+		for i := range r {
+			if i < proofsLen {
+				r[i] = rideBytes(common.Dup(proofs.Proofs[i].Bytes()))
+			} else {
+				r[i] = rideBytes(nil)
+			}
+		}
+	} else {
+		// special case for ethereum stuff
+		for i := range r {
```

Comment:
I think better to move this to `r` initialisation. Looks like it's OK to have this not only for eth transactions and actually it works now not only for eth transactions.